### ISSUE TITLE
Question: What's is the roadmap for K?

### ISSUE THREAD

**Issue by Camille92** [ctubio/Krypto-trading-bot#40]
Hi Carles,

More of a general question here, what's your roadmap for K?

And how can I help? Do you need funds and how much? Can I help by helping for strategies/ways of implementing?

Best,

Camille

**Comment by ctubio** [comment]
## TL;DR: once K have 0 bugs, if ppl keeps making 0 donations, ima impose a fee (ofcourse using bitcoins and automatic server-less payment verification :dancer:)

my plan is to migrate all to C++, make it faster and smaller. well really meanwhile this happens, im trying to fix all the bugs and memory leaks (this application deserves to be excellent [because it have to do a critical mission] but currently is a factory of bugs).

now that we succeed with basic math and database, next is the websockets, then the gateways logic and finally the quoting engine logic; then i can go to sleep :D (ok well, also once all is C++, then include support for winy installation).

as you see my roadmap does not include any new strategy; so all suggestions about strategies are very welcome. (i dont plan to make more complex the app until migrated all to C++, but ofcourse thats only my initial intention, suggestions are very appreciated and always considered'¡)

i really feel bad asking for funds; when i do is really because i feel abused.
you already send many funds last month so i double ask you to dont make me feel double bad sending more (until all your already openend new pending features are implemented at least xD) xD

aaalso, since i dont like to ask for funds, and since nobody donates (this is not true, 3 lovely users donated and i feel very very happy and proud about it; but im comparing to the hundreds that downloaded it), in the roadmap is also the plan to make all this new C++ features commercial (with a fee to unlock them), and leave the original javascript version as a free version. i will only offer the C++ version for free if i receive enough donations by the time i finnish.

if ever fees are imposed, will be small; just for fun (i believe everybody have fun making small coin transactions, so lets all have fun!) not like crazy expensive https://bitcointalk.org/index.php?topic=507103.0 (i just give you the link just for the records, not as a suggestion because i can tell this trader is a piece of shit xD)

aaaaaalso if you ask me, i would like to make some social site, (i hate social sites), but to share quoting parameters or im not sure what ppl would like to share about the app, but im sure there is something to share, so a site for it seems needed; i just don't know the propose of it yet xD (<<< somehow, but all anonymous, nothing about registrations or emals or usernames; ultimately if all ideas fail i will simply open a thread in some bitcointalk subforum to allow more topical discussions than simply technical discussions in github; i didnt checked since long, but i think in that forum im still marked as scammer; i mean, there are not many scammers as a project maintainers in bitcointalk, we could have some fun xD)

P.D. no, i didnt scam nobody; i was having fun making a mning pool xD and i send back the money lost to the affected user; but i like the mark if you ask me :P (and i end up making 3 mining pools in 3 different languages, so i regret nothing :D)

**Comment by Camille92** [comment]
thank you for the details, yep I agree it will be lovely to be bug-free haha!

I think the forum should be a good idea, a guy in Michael's repo just proposed that haha. Is there's not some free software about anonymous (or nearly) software?
I don't mind a small registration with username and password (and email if you wish for backup) so that I'm very sure to who I'm talking too. (I mean anonymity on the internet is the best AND the worst thing).

Yes why not a small fee it's worth it haha!

**Comment by Camille92** [comment]
Btw about the downloads don't you think it can come from the git pull we do as well?

And it counts each time I (or anyone) run a clean install of K. Maybe the actual number is smaller :)

**Comment by zachaller** [comment]
A suggestion of maybe a reddit sub for the forum it's free really popular and lots of people already have accounts.

**Comment by ctubio** [comment]
downloads currently come from https://www.npmjs.com/package/hacktimer that is only included in this app; git doesn't have a count of pulls

**Comment by CarlitoGrey** [comment]
Reddit maybe a quick and easy way to go.
